### PR TITLE
MES-2295: Get RDS IAM config + save result to Aurora

### PR DESCRIPTION
**Pull request by johnmorleybjss**
# Description and relevant Jira numbers
* Get IAM credentials so we can access RDS (code from the pollers)
* Build the insert queries for `TEST_RESULT` and `UPLOAD_QUEUE` tables
* Populate parameters for the inserts from the test result model
* Save the 4 records in a transaction with rollbacks on errors
* Still needs some tests for the more integration heavy parts, could well be in a subsequent PR

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [x] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

File: src/functions/postResult/application/save-result-service.ts [hunk 1]
```diff
@@ -1,5 +1,48 @@
 import { StandardCarTestCATBSchema } from '@dvsa/mes-test-schema/categories/B';
+import * as mysql from 'mysql2';
+import { IntegrationType } from '../domain/result-integration';
+import { getConnection } from '../../../common/framework/mysql/database';
+import { buildTestResultInsert, buildUploadQueueInsert } from '../framework/database/query-builder';
 
-export const saveTestResult = (testResult: StandardCarTestCATBSchema): Promise<void> => {
-  return Promise.resolve();
+export const saveTestResult = async (testResult: StandardCarTestCATBSchema): Promise<void> => {
+  const connection: mysql.Connection = getConnection();
```
Comment (rstepnowski) [inline src/functions/postResult/application/save-result-service.ts]: Should we close this connection at some point?   
Are we planning to introduce the connection pooling in this lambda (similar to the pooler one) or we're happy to just manually open/close connections?

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "compilerOptions": {
     "outDir": "./build",
-    "strict": true,
```
Comment (rstepnowski) [inline tsconfig.json]: What was the reason for removing strict checking?

File: src/functions/postResult/application/save-result-service.ts [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+  const uploadQueueInsertTars = buildUploadQueueInsert(testResult, IntegrationType.TARS);
+  const uploadQueueInsertRsis = buildUploadQueueInsert(testResult, IntegrationType.RSIS);
+  const uploadQueueInsertNotify = buildUploadQueueInsert(testResult, IntegrationType.NOTIFY);
+
+  return new Promise((resolve, reject) => {
+    connection.beginTransaction((err) => {
+      if (err) {
+        reject(err);
+      }
+      connection.query(testResultInsert, (err) => {
```
Comment (ammarhaiderbjss) [inline src/functions/postResult/application/save-result-service.ts]: How comes you're not using your `query` function from `src/common/framework/mysql/database.ts`?

File: tsconfig.json (hunk 2 above)
Comment (johnmorleybjss) [reply tsconfig.json]: This is something that we also had to do with the pollers. In order for AWS IAM auth to work, we need the `mysql2` package, but this package doesn't ship with the correct type defs, causing TS compilation to fail.
```
Could not find a declaration file for module 'mysql2'
```

File: src/functions/postResult/application/save-result-service.ts (hunk 3 above)
Comment (johnmorleybjss) [reply src/functions/postResult/application/save-result-service.ts]: Ended up following the docs for transactions on the `mysql` driver docs. I will try using the helper method to clean this up.

File: src/functions/postResult/application/save-result-service.ts (hunk 1 above)
Comment (johnmorleybjss) [reply src/functions/postResult/application/save-result-service.ts]: Added closing of connections.

Comment (johnmorleybjss) [reply src/functions/postResult/application/save-result-service.ts]: Don't see a reason for a connection pool in this Lambda, as everything that we're saving needs to be bounded by the same transaction - multiple connections wouldn't help us here.

**Pull request merged**